src/parser.ts

```diff
@@ -40,9 +40,9 @@ const registryDocumentMap = new WeakMap<Document, Registry<ParserObjectConstruct
 const nodeMap = new WeakMap<HTMLElement, ParserObject>();
 
 /**
- * A hash of the object ids and their objects
+ * A map of document objects to a hash of the object ids and their objects
  */
-const idMap: { [id: string]: ParserObject } = {};
+const idDocumentMap = new WeakMap<Document, { [id: string]: ParserObject }>();
```

Comment:
Since we are using our idMap hash in several places now, maybe best to extract it to an interface.